test/e2e/util_test.go

```diff
@@ -983,12 +984,81 @@ func isFeatureGateEnabled(featureGateName configv1.FeatureGateName) (bool, error
+	serviceName := controller.LoadBalancerServiceName(ic)
+	if err := kclient.Get(context.Background(), serviceName, lbService); err != nil {
+		return fmt.Errorf("failed to get service: %v", err)
+	}
+	oldUid := lbService.UID
+
+	if err := kclient.Delete(context.Background(), lbService); err != nil {
+		return fmt.Errorf("failed to delete service: %v", err)
+	}
+
+	// Wait for the service to be recreated by Ingress Operator.
+	err := wait.PollUntilContextTimeout(context.Background(), 10*time.Second, 3*time.Minute, false, func(ctx context.Context) (bool, error) {
+		if err := kclient.Get(context.Background(), controller.LoadBalancerServiceName(ic), lbService); err != nil {
+			t.Logf("failed to get service %s/%s: %v, retrying ...", serviceName.Namespace, serviceName.Namespace, err)
+			return false, nil
+		}
+		if oldUid == lbService.UID {
+			t.Logf("service %s/%s UID has not changed yet, retrying...", serviceName.Namespace, serviceName.Namespace)
+			return false, nil
+		}
+		return true, nil
+	})
+	return err
+}
+
+// classicLoadBalancerParametersSpecExist checks if ClassicLoadBalancerParameters exist in the IngressController spec.
+func classicLoadBalancerParametersSpecExist(ic *operatorv1.IngressController) bool {
+	return ic.Spec.EndpointPublishingStrategy != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil
+}
+
+// networkLoadBalancerParametersSpecExists checks if NetworkLoadBalancerParameters exist in the IngressController spec.
+func networkLoadBalancerParametersSpecExists(ic *operatorv1.IngressController) bool {
+	return ic.Spec.EndpointPublishingStrategy != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.NetworkLoadBalancerParameters != nil
+}
+
+// networkLoadBalancerParametersStatusExist checks if NetworkLoadBalancerParameters exist in the IngressController
+// status.
+func networkLoadBalancerParametersStatusExist(ic *operatorv1.IngressController) bool {
+	return ic.Status.EndpointPublishingStrategy != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.NetworkLoadBalancerParameters != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.NetworkLoadBalancerParameters.Subnets != nil
```

Comment:
do you need to check this condition for Subnets ?

Reply:
Whoops, no, thanks! Fixed.